jenkins-plugin/src/test/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepNoOptionalsTest.java

```diff
@@ -0,0 +1,99 @@
+package org.jenkinsci.plugins.pipeline.maven;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+
+import org.jenkinsci.plugins.pipeline.maven.publishers.PipelineGraphPublisher;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.remoting.RoleChecker;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.InboundAgentRule;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.RealJenkinsRule;
+
+import hudson.FilePath;
+import hudson.model.Result;
+import hudson.model.Slave;
+import hudson.remoting.Callable;
+import hudson.tasks.Maven;
+import jenkins.model.Jenkins;
+import jenkins.mvn.DefaultGlobalSettingsProvider;
+import jenkins.mvn.DefaultSettingsProvider;
+import jenkins.mvn.GlobalMavenConfig;
+
+public class WithMavenStepNoOptionalsTest {
+
+    @Rule
+    public RealJenkinsRule jenkinsRule = new RealJenkinsRule()
+        // .withLogger("org.jenkinsci.plugins.pipeline.maven", Level.ALL)
+        .omitPlugins(
+            "analysis-core", "bootstrap5-api", "checks-api", "commons-lang3-api", "commons-text-api", "configuration-as-code",
+            "echarts-api", "font-awesome-api", "mysql-api", "postgresql-api", "maven-plugin", "flaky-test-handler",
+            "htmlpublisher", "jacoco", "jgiven", "junit", "junit-attachments", "matrix-project",
+            "maven-invoker-plugin", "pipeline-build-step", "plugin-util-api", "findbugs", "tasks");
+
+    @ClassRule
+    public static InboundAgentRule agentRule = new InboundAgentRule();
+
+    @Test
+    public void maven_build_jar_project_on_master_succeeds() throws Throwable {
+        jenkinsRule.then(j -> setup(j), j -> build(j));
+    }
+
+    private static void build(JenkinsRule r) throws Throwable {
+        String pipelineScript = "node('mock') {\n" +
+                "    git 'https://github.com/jglick/simple-maven-project-with-tests'\n" +
```

Comment:
BTW while I have no plans to ever delete this repo, or host malicious content from it, it would be better not to trust me!

Reply:
I agree. I ended with this because of master/slave, I did not get that they share the same filesystem, so I could use local repo, like other tests

Reply:
Yeah, so long as you are not using a _Dockerized_ agent you can just assume file paths good on the test & controller are also good on the agent.